src/System.Net.Http/src/System/Net/Http/Managed/ConnectHelper.cs

```diff
@@ -22,10 +22,7 @@ internal static class ConnectHelper
                 // to cancel it if needed.
                 using (var saea = new BuilderAndCancellationTokenSocketAsyncEventArgs(cancellationToken))
                 {
-                    // Configure which server to which to connect.
-                    saea.RemoteEndPoint = IPAddress.TryParse(host, out IPAddress address) ?
-                        (EndPoint)new IPEndPoint(address, port) :
-                        new DnsEndPoint(host, port);
+                    saea.RemoteEndPoint = new DnsEndPoint(host, port);
```

Comment:
While the change works, if an address is specified, doesn't that end up taking a non-trivially more expensive code path? Before this is changed, it'd be good to look at a profile (including allocations); if I'm right, maybe we should still take the change and fix it in Socket, but I wouldn't want to regress this when the original code was pretty simple.

Reply:
Yeah, you are correct here.  I should have looked at the ConnectAsync code more closely.

We should consider fixing this in Socket, but that seems like a more involved change.

I'll revert and file a separate issue.

Reply:
Filed #26517